src/script/event/EventRepository.js

```diff
@@ -257,64 +257,72 @@ export class EventRepository {
    * @param {number} [limit=EventRepository.CONFIG.NOTIFICATION_BATCHES.MAX] Max. number of notifications to retrieve from backend at once
    * @returns {Promise} Resolves when all new notifications from the stream have been handled
    */
-  getNotifications(notificationId, limit = EventRepository.CONFIG.NOTIFICATION_BATCHES.MAX) {
-    return new Promise(async (resolve, reject) => {
-      const _gotNotifications = ({has_more: hasAdditionalNotifications, notifications, time}) => {
-        if (time) {
-          this.serverTimeHandler.computeTimeOffset(time);
-        }
-
-        if (notifications.length > 0) {
-          notificationId = notifications[notifications.length - 1].id;
+  async getNotifications(notificationId, limit = EventRepository.CONFIG.NOTIFICATION_BATCHES.MAX) {
+    const processNotifications = notifications => {
+      if (notifications.length <= 0) {
+        this.logger.info(`No notifications found since '${notificationId}'`);
+        this.notificationsLoaded(true);
+        return notificationId;
+      }
 
-          this.logger.info(`Added '${notifications.length}' notifications to the queue`);
-          this.notificationsQueue.push(...notifications);
+      notificationId = notifications[notifications.length - 1].id;
 
-          if (!this.notificationsPromises) {
-            this.notificationsPromises = [resolve, reject];
-          }
+      this.logger.info(`Added '${notifications.length}' notifications to the queue`);
+      this.notificationsQueue.push(...notifications);
 
-          this.notificationsTotal += notifications.length;
+      this.notificationsTotal += notifications.length;
 
-          // FIXME: Just one more call to /notifications might not be enough. Consider using "getAllNotificationsForClient" from "NotificationsService".
-          if (hasAdditionalNotifications) {
-            return this.getNotifications(notificationId, EventRepository.CONFIG.NOTIFICATION_BATCHES.SUBSEQUENT);
-          }
+      this.notificationsLoaded(true);
+      this.logger.info(`Fetched '${this.notificationsTotal}' notifications from the backend`);
+      return notificationId;
+    };
 
-          this.notificationsLoaded(true);
-          this.logger.info(`Fetched '${this.notificationsTotal}' notifications from the backend`);
-          return notificationId;
+    await this._handleTimeDrift();
+
+    try {
+      const notificationList = await this.notificationService.getAllNotificationsForClient(
+        this.currentClient().id,
+        notificationId,
+        limit,
+      );
+      // Note: `resolve` handle is needed to delay notification handling until buffered notifications from websocket are processed.
+      return new Promise((resolve, reject) => {
```

Comment:
This is weird stuff here, it will be gone with the api-client websocket integration